components/service/glean/scripts/sdk_generator.gradle

```diff
@@ -323,17 +329,30 @@ if (project.ext.has("gleanGenerateMarkdownDocs")) {
         }
     }
 
-    // Attach the docs generation task to the code generation task, to make sure
-    // we generate docs, if we're asked for it.
-    tasks.whenTaskAdded { task ->
-        if (task.name.startsWith(gleanTaskNamePrefix)) {
-            task.dependsOn(gleanGenerateMetricsDocs)
+    // Only attach the generation task if the metrics file is available or we're requested
+    // to fetch them from AAR files. We don't need to fail hard otherwise, as some 3rd party
+    // project might just want metrics included in Glean and nothing more.
+    if (file("$projectDir/metrics.yaml").exists()
+        || file("$projectDir/pings.yaml").exists()
+        || project.ext.has("allowMetricsFromAAR")) {
+        // Generate the Metrics docs, if requested.
+        if (project.ext.has("gleanGenerateMarkdownDocs")) {
+            // Attach the docs generation task to the code generation task, to make sure
+            // we generate docs, if we're asked for it.
+            generateKotlinAPI.dependsOn(generateDocsTask)
```

Comment:
I think the source task is run a number of times for `release`, `debug` etc.  It's probably fine to also build the docs multiple times here, just wanted to call that out.  This is one of those things where I would say "good enough for Gradle", because fixing it might be way more effort than it's worth...

Reply:
Yes, unfortunately we had the same behaviour before. Due to the way we're building the task, I failed at making this a "thing we only run once across al variants" :(

Reply:
Ok.  It's not like it's a long-running task, at least. :)